webapi/src/main/scala/org/knora/webapi/messages/StringFormatter.scala

```diff
@@ -2007,49 +2034,63 @@ class StringFormatter private (
       case None    => None
     }
 
+  def validateAndEscapeOptionalProjectShortname(
+    maybeString: Option[String]
+  ): Validation[ValidationException, Option[String]] =
+    maybeString match {
+      case None        => Validation.succeed(None)
+      case Some(value) => validateAndEscapeProjectShortname(value).map(Some(_))
+    }
+
   /**
    * Given the project shortcode, checks if it is in a valid format, and converts it to upper case.
    *
    * @param shortcode the project's shortcode.
    * @return the shortcode in upper case.
    */
+  @deprecated("Use def validateProjectShortcode(String) instead.")
   def validateProjectShortcode(shortcode: String, errorFun: => Nothing): String = // V2 / value objects
-    ProjectIDRegex.findFirstIn(shortcode.toUpperCase) match {
-      case Some(value) => value
-      case None        => errorFun
-    }
+    validateProjectShortcode(shortcode).getOrElse(errorFun)
+
+  def validateProjectShortcode(shortcode: String): Option[String] =
+    ProjectIDRegex.findFirstIn(shortcode.toUpperCase)
 
+  @deprecated("Use escapeOptionalString(Option[String]) instead.")
   def escapeOptionalString(maybeString: Option[String], errorFun: => Nothing): Option[String] = // --
     // TODO: I leave this for now to avoid merge conflicts. Should be moved to the ValuesValidator as soon as possible. (depends on toSparqlEncodedString())
-    maybeString match {
-      case Some(s) =>
-        Some(toSparqlEncodedString(s, errorFun))
-      case None => None
-    }
+    maybeString.map(toSparqlEncodedString).getOrElse(errorFun)
+
+  def escapeOptionalString(maybeString: Option[String]): Option[String] =
+    maybeString.flatMap(toSparqlEncodedString)
 
   /**
    * Given the group IRI, checks if it is in a valid format.
    *
    * @param iri the group's IRI.
    * @return the IRI of the list.
    */
+  @deprecated("Use validateGroupIri(IRI) instead.")
   def validateGroupIri(iri: IRI, errorFun: => Nothing): IRI = // V2 / value objects
-    if (isKnoraGroupIriStr(iri)) {
-      iri
-    } else {
-      errorFun
-    }
+    validateGroupIri(iri).getOrElse(errorFun)
+
+  def validateGroupIri(iri: IRI): Validation[ValidationException, IRI] =
+    if (isKnoraGroupIriStr(iri)) Validation.succeed(iri)
+    else Validation.fail(ValidationException(s"Invalid IRI: $iri"))
 
   /**
    * Given the optional group IRI, checks if it is in a valid format.
    *
    * @param maybeIri the optional group's IRI to be checked.
    * @return the same optional IRI.
    */
+  @deprecated("Use validateOptionalGroupIri(Option[IRI]) instead.")
   def validateOptionalGroupIri(maybeIri: Option[IRI], errorFun: => Nothing): Option[IRI] = // V2 / value objects
+    validateOptionalGroupIri(maybeIri).getOrElse(errorFun)
+
+  def validateOptionalGroupIri(maybeIri: Option[IRI]): Validation[ValidationException, Option[IRI]] =
```

Comment:
Consider removing this method and inline the original `validateOptionalGroupIri` instead.

Reply:
All "optional" methods are gone now 🔥